lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -88,7 +88,18 @@ defmodule Trento.Domain.SapSystem do
     embeds_one :application, Application
   end
 
+  def execute(
+        %SapSystem{},
+        %RegisterDatabaseInstance{
+          system_replication: "Secondary",
+          system_replication_status: "ACTIVE"
+        }
+      ),
+      do: {:error, :database_instance_not_primary}
```

Comment:
The error should be `:sap_system_not_registered` to indicate that an instance was discovered but no sap system was registered yet. Keep in mind that those domain errors are then bubbling up to the fallback controller, in this case we return a 404.

Reply:
Right, fix coming asap

Reply:
done